### PR TITLE
return Response Header when caller set returnResponseHeader to true i…

### PR DESCRIPTION
**Pull request by jennifersalesforce**
@stomita 
Would you please review my pull request?
I want to return request headers for query if option.returnResponseHeader is true.

### CODE REVIEW

File: lib/connection.js [hunk 1]
```diff
@@ -310,16 +310,21 @@ Connection.prototype.request = function(request, options, callback) {
...
+            if (apiUsage) {
+                self.limitInfo = {
+                    apiUsage: {
+                        used: parseInt(apiUsage[1], 10),
+                        limit: parseInt(apiUsage[2], 10)
+                    }
+                };
+            }
+        }
+        if (options.returnResponseHeaders) {
```
Comment (jennifersalesforce) [inline lib/connection.js]: return responseHeaders when options.returnResponseHeaders is true.

**Comment by jennifersalesforce** [comment]
@stomita 
Would you please review my pull request and let me know if I need to change anything?

Thanks 
Jennifer

**Comment by jennifersalesforce** [comment]
@stomita @joeferraro @stanlemon @icecubed 
Would you please review my change to see if I can merge?

Thanks
Jennifer

**Comment by tnoonan-salesforce** [comment]
@stomita Poke.

@jennifersalesforce Maybe add @stomita to the reviewer list.

**Comment by jennifersalesforce** [comment]
@stomita 
Looks like I don't have permission to add reviewers.
Would you please help to review it?

File: lib/connection.js (hunk 1 above)
Comment (amphro) [reply lib/connection.js]: I think this needs to be reworked. First, `self` is the connection, not the request. So you could be dealing with a response that is not from your request. Second, why do we need to set an option to get the response headers? Instead, I would like to see this as part of the httpApi that is passed with the callback.

**Comment by stomita** [comment]
@jennifersalesforce 
I'm against for storing response headers in connection properties, it is not the proper place.

**Pull request closed without merging**